js_modules/dagit/packages/core/src/runs/DeletionDialog.tsx

```diff
@@ -24,11 +19,14 @@ export interface Props {
 
 type SelectedRuns = {[id: string]: boolean};
 
-type Error =
-  | Delete_deletePipelineRun_PythonError
-  | Delete_deletePipelineRun_UnauthorizedError
-  | Delete_deletePipelineRun_RunNotFoundError
-  | undefined;
+const refineToError = (data: DeleteMutation | null | undefined) => {
+  if (data?.deletePipelineRun.__typename === 'DeletePipelineRunSuccess') {
+    throw new Error('Not an error!');
+  }
+  return data?.deletePipelineRun;
+};
+
+type Error = ReturnType<typeof refineToError>;
```

Comment:
I added a handful of these refinement functions in places where we previously had a manually-defined type union for non-success states. This allows us to avoid pulling in a bunch of types or defining fragments for each error-ish member of the union.